From: Akihiro Sagawa <sagawa.aki@gmail.com>
Subject: [PATCH v2 2/2] kernel32/tests: Add mutex tests with 8-bit character string.
Message-Id: <20200520214354.884C.375B48EC@gmail.com>
Date: Wed, 20 May 2020 21:47:10 +0900


v2: Simplify the tests and swap the patche order.

We should be able to open the mutex with the same name after creation.

Signed-off-by: Akihiro Sagawa <sagawa.aki@gmail.com>
---
 dlls/kernel32/tests/sync.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/dlls/kernel32/tests/sync.c b/dlls/kernel32/tests/sync.c
index 0f072a5846e..465e6550d03 100644
--- a/dlls/kernel32/tests/sync.c
+++ b/dlls/kernel32/tests/sync.c
@@ -306,6 +306,17 @@ todo_wine_if(getenv("WINEESYNC"))   /* XFAIL: due to the above */
 
     CloseHandle(hCreated);
 
+    /* Test 8-bit character string */
+
+    hCreated = CreateMutexA(NULL, FALSE, "abcd\xef");
+    ok(hCreated != NULL, "CreateMutex failed with error %d\n", GetLastError());
+
+    hOpened = OpenMutexA(SYNCHRONIZE, FALSE, "abcd\xef");
+    ok(hOpened != NULL, "OpenMutexA failed with error %d\n", GetLastError());
+    CloseHandle(hOpened);
+
+    CloseHandle(hCreated);
+
     mutex = CreateMutexA( NULL, FALSE, NULL );
     ok(!!mutex, "got error %u\n", GetLastError());
 
